go/kbfs/libkbfs/prefetcher.go

```diff
@@ -812,7 +814,12 @@ func (p *blockPrefetcher) run(testSyncCh <-chan struct{}) {
 	}()
 	isShuttingDown := false
 	var shuttingDownCh <-chan interface{}
+	first := true
 	for {
+		if !first && testDoneCh != nil && !isShuttingDown {
+			testDoneCh <- struct{}{}
```

Comment:
A single channel doesn't given the test enough control, because it can't serialize the work of the prefetcher and the maximizing the disk usage in the test itself.  Specifically, the situation it addresses is:

1. Set the limiter limits to equal the current cache size.
2. One send on `testSyncCh` releases a queued prefetch processing of a block.
3. The test then makes more room in the limiter.

The problem is that step 3 can happen _during_ step 2, since with only one channel we have no way of knowing when step 2 completed.  If that happens, the test assumptions will be wrong, because the prefetcher will be unable to write the block and its metadata into the sync cache (because the limiter says there's no room).

Reply:
Oh I see, thanks. Please add a comment to that effect.